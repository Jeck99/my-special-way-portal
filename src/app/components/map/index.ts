export * from './map.component';
export * from './paths-mapper.pipe';
