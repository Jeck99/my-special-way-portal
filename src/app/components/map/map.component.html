<button type="button" 
        *ngFor="let floor of availableFloors" 
        (click)="setFloor(floor)">floor {{floor}}</button>
<div class ="map"
    leaflet 
    [leafletOptions]="options"
    [leafletLayers]="indoorAtlasPaths | pathsMapper:currentFloor"
    [leafletCenter]="center">
</div>