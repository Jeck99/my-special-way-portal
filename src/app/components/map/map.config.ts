
import { latLng, tileLayer } from 'leaflet';

export const DEFAULT_POSITION = {
    latitude: 31.986422990470114,
    longtitude: 34.912109971046455,
};

export const DEFAULT_MAP_OPTIONS = {
    layers: [
        tileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', { maxZoom: 18 }),
    ],
    zoom: 18,
    center: latLng(DEFAULT_POSITION.latitude, DEFAULT_POSITION.longtitude),
};
