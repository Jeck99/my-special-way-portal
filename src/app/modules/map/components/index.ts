export * from './map.component';
export * from './pipes/paths-mapper.pipe';
export * from './services/map.service';
