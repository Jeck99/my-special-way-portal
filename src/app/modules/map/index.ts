export * from './map.module';
export * from './components';
