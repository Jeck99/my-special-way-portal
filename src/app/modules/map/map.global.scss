@import  "../node_modules/leaflet/dist/leaflet.css";
@import "../node_modules/leaflet-draw/dist/leaflet.draw.css";

.my-beautiful-icon {
    background: rgb(56, 155, 255) !important;
    border-radius: 50%;
}