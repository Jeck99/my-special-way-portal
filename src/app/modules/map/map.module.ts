import { LeafletModule } from '@asymmetrik/ngx-leaflet';
import { LeafletDrawModule } from '@asymmetrik/ngx-leaflet-draw';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import {MatRadioModule} from '@angular/material/radio';

import {
    MapComponent,
    PathsMapperPipe,
    MapService,
} from './components';

@NgModule({
    imports: [
        LeafletModule.forRoot(),
        LeafletDrawModule.forRoot(),
        CommonModule,
        MatRadioModule,
    ],
    declarations: [
        MapComponent,
        PathsMapperPipe,
    ],
    providers: [
        MapService,
    ],
})
export class MapModule {
}
