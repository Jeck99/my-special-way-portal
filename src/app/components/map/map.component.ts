import { Component, OnInit } from '@angular/core';
import { LatLng, latLng } from 'leaflet';
import { DEFAULT_MAP_OPTIONS } from './map.config';
import { PATHS } from './paths.mock';
import { IndoorAtlasPaths, LeafletPaths } from './map.model';
@Component({
    selector: 'app-map',
    templateUrl: './map.component.html',
    styleUrls: ['./map.component.scss'],
})
export class MapComponent implements OnInit {
    options = DEFAULT_MAP_OPTIONS;
    center: LatLng | undefined;
    currentFloor = 1;
    layers: LeafletPaths;
    availableFloors: number[];
    indoorAtlasPaths: IndoorAtlasPaths = PATHS;

    ngOnInit() {
        this.availableFloors = this.getAvailableFloors();
        if (navigator.geolocation) {
            navigator.geolocation.getCurrentPosition((position) => {
                this.center = latLng(position.coords.latitude, position.coords.longitude);
            });
        }
    }

    setFloor(floorNum: number) {
        this.currentFloor = floorNum;
    }

    private getAvailableFloors() {
        return PATHS.nodes.reduce((floors, node) => {
            if (floors.indexOf(node.floor) === -1) {
                floors.push(node.floor);
            }
            return floors;
        }, [] as number[]).sort();
    }
}
