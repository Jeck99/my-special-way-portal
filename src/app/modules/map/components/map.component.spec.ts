import { TestBed } from '@angular/core/testing';
import { MapComponent, PathsMapperPipe } from './index';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { MapService } from './map.service';
import { Observable } from 'rxjs/Observable';

const mapMock = {
    getAllPaths: () => Observable.of({nodes: [], edges: []}),
    getAllAvailableFloors: () => Observable.of([2, 3 ]),
    getAllMapWayPoints: () => Observable.of([]),
    getCurrentPosition: () => Promise.resolve(),
};

describe('map component', () => {
    beforeEach(async () => {
        TestBed.configureTestingModule({
            declarations: [
                MapComponent,
                PathsMapperPipe,
            ],
            providers: [
                {provide: MapService, useValue: mapMock},
            ],
            schemas: [NO_ERRORS_SCHEMA],
          });
    });

    it('should render the map component correctly', () => {
        const fixture = TestBed.createComponent(MapComponent);
        expect(fixture).toMatchSnapshot();
    });

    // it('should display paths from floor 2 when clicking on floor 2 button', () => {
    //     const fixture = TestBed.createComponent(MapComponent);
    //     const floor3Button = fixture.debugElement.query(By.css('button'));
    //     floor3Button.click();

    //     // this isn't the best way to test it, I need to figure out a better way
    //     expect(fixture.componentInstance.currentFloor).toEqual(2);
    // });
});
