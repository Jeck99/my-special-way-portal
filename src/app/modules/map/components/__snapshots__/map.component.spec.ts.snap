// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map component should render the map component correctly 1`] = `
<app-map
  availableFloors={[Function Array]}
  currentFloor={[Function Number]}
  indoorAtlasPaths={[Function Object]}
  mapService={[Function Object]}
  options={[Function Object]}
>
  
  <div
    class="map"
    leaflet=""
  >
    
  
  </div>
</app-map>
`;
